### PR TITLE
Inline Help: Add an onboarding welcome at the preview page.

### PR DESCRIPTION
**Pull request by ramonjd**
## Changes proposed in this Pull Request

screen shot 2019-01-29 at 4 15 51 pm

Testing out whether we can/should open the inline popup with onboarding welcome content via a query param.

### Considerations ❓ 
The query param `welcome` triggers the content. Should we program in some limitation to this, for example, the onboarding inline-help content should only show, despite the presence of the `welcome` parameter when:
1. All checklist items **have not** been completed
2. The user has created the current site in the last minute, or has just been redirected from the onboarding flow

~Also, the design omits regular help content (Help/contact links and search) from the bubble. Should we include something to indicate to the user that help is available, to ensure that they know the intended purpose of inline help?~ We've also added a Help link to the footer.

## Testing instructions

1. Create a new site
2. When you're redirected to preview, check that the URL contains the `?welcome` parameter
http://calypso.localhost:3000/view/yoursite.wordpress.com?welcome The inline-help with welcome message should be present
3. Click **Contact us**, then click **Back**. The inline-help with welcome message should be present
3. Close by clicking **Not now**, then reopen the inline help bubble. The regular help content (Help link, search etc) should show.
4. Refresh the page to show the congrats message again, and click **< Help**. The bubble content should display the regular help content plus checklist progress widget.
5. Refresh the page to show the congrats message again, and click **Start Customizing**. You should be redirected to the checklist, and reopening the inline help bubble should display regular help content .

### CODE REVIEW

**Comment by mattwiebe** [comment]
Fixed up the merge conflict and tested this. Code looks good and works as expected in testing instructions. I see that we might want to add a `< Help` button yet too to return to the main help portion of the popover so I won't give this the "official" 👍 review yet.

**Comment by ramonjd** [comment]
@danhauk @fditrapani 

Added **Help** and **Contact** buttons to the footer

jan-29-2019 16-01-55

**Comment by fditrapani** [comment]
Nice! Looks good and tests well. The pop up shows after my site is created. One thing I noticed is that it blows up if you press the contact button. This is what I see:

image

**Comment by danhauk** [comment]
This looks great, thanks @ramonjd! I see the same thing as @fditrapani noted. It looks like clicking `Contact us` from the welcome message isn't showing the contact content. Clicking it from the regular inline help screen shows it as expected:

image

**Comment by ramonjd** [comment]
🤦‍♂️ 

@fditrapani @danhauk Thanks so much for the 👀 Gave me the chance to refactor the logic a bit to be more consistent with the secondaryView constants approach.


jan-30-2019 11-56-16

File: client/blocks/inline-help/popover.jsx [hunk 1]
```diff
@@ -122,15 +219,16 @@ class InlineHelpPopover extends Component {
...
-								setDialogState={ this.props.setDialogState }
-								closePopover={ this.props.onClose }
+								result={ selectedResult }
+								setDialogState={ setDialogState }
+								closePopover={ onClose }
 							/>
 						),
-						checklist: <WpcomChecklist closePopover={ this.props.onClose } viewMode="prompt" />,
+						[ VIEW_CHECKLIST ]: <WpcomChecklist closePopover={ onClose } viewMode="prompt" />,
+						[ VIEW_ONBOARDING_WELCOME ]: <ChecklistOnboardingWelcome onClose={ onClose } />,
```
Comment (ramonjd) [inline client/blocks/inline-help/popover.jsx]: _Explanation_: We may as well use the constant values as the object keys to reduce the chance of typos.

**Comment by adamziel** [comment]
This essentialy replaces the inline checklist feature that's available via `/view?help&checklist` - is this what we want to do here? I thought of the welcome screen more as the first step of the inline checklist rather than replacement that would link to `/checklist`. Or is it just for now until we're finished with new tasks?

**Comment by adamziel** [comment]
cc @fditrapani @michaeldcain

**Comment by ramonjd** [comment]
> Yes, the Start customizing button should go to the first task, not the checklist. 

Thanks for the clarification. I'll create a card for this task (if one hasn't been created already). 👍

**Pull request merged**